### PR TITLE
Test new export flow: e2e tests to verify 10G/210G/1T/8T

### PR DESCRIPTION
**Pull request by dntczdx**
There are overall 3 changes:
1. improvement of workflow: delete disk resource proactively. Without this change, disk resource will keep alive during the whole testing, which may exhaust quota.
2. test preparation workflow: it's done by files started with "create_random_disk_*". It creates a image in project "compute-image-tools-test" with full-filled random data, which will be used in the e2e test. It's a one-time preparation. We don't want to populate random data during the e2e test runnung, because it's quite time consuming.
3. actual e2e test workflow: it exports the image to vmdk format, and then verifies by import && mount && ls.

This test set covers size of 10GB (fast and easy), 210GB (doesn't work in old workflow but supposely work fine in new workflow), 1TB (a reasonable large disk), and 8TB (a large disk that we theoretically can support).

Exporting a large image without format converting is not covered in this CL due that logic is not changed recently. I will cover it later as a lower priority.

### CODE REVIEW

File: daisy_integration_tests/create_random_data_disk.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+i# Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (zoran15) [inline daisy_integration_tests/create_random_data_disk.sh]: 2019

Comment (dntczdx) [reply daisy_integration_tests/create_random_data_disk.sh]: resolved

**Comment by dntczdx** [comment]
updated typo

File: daisy_integration_tests/create_random_data_disk.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+BYTES_1GB=1073741824
+METADATA_URL="http://metadata/computeMetadata/v1/instance"
+DISK_DATA_NAME=$(curl -f -H Metadata-Flavor:Google ${METADATA_URL}/attributes/disk-data-name)
+SIZE=$(curl -f -H Metadata-Flavor:Google ${METADATA_URL}/attributes/disk-size)
+IMAGE_DESTINATION=$(curl -f -H Metadata-Flavor:Google ${METADATA_URL}/attributes/image-destination)
+ZONE=$(curl -f -H Metadata-Flavor:Google ${METADATA_URL}/zone | cut -d'/' -f4)
+RESERVED_SPACE=50 #50MB reserved space so system can still run
+
+echo "GCECreateDisk: Checking whether image has been created."
```
Comment (adjackura) [inline daisy_integration_tests/create_random_data_disk.sh]: This check is already done by daisy, this script shouldn't need any of this logic

File: daisy_integration_tests/create_random_data_image.subwf.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	    "disk-data-name": "disk-data-${NAME}-${ID}",
+	    "image-destination": "image-export-given-size-${size}gb"
+          },
+          "networkInterfaces": [
+            {
+              "network": "${export_network}",
+              "subnetwork": "${export_subnet}"
+            }
+          ],
+          "Scopes": ["https://www.googleapis.com/auth/devstorage.full_control", "https://www.googleapis.com/auth/compute"],
```
Comment (adjackura) [inline daisy_integration_tests/create_random_data_image.subwf.json]: For this simple script you should need no special scopes, its just filling up a disk then shuting down

File: daisy_integration_tests/create_random_data_image.subwf.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  "Name": "create-random-data-image",
+  "Vars": {
+    "size": {
+      "Required": true
+    },
+    "export_instance_disk_image": {
+      "Value": "projects/compute-image-tools/global/images/family/debian-9-worker",
+      "Description": "image to use for the worker instance"
+    },
+    "export_network": {
```
Comment (adjackura) [inline daisy_integration_tests/create_random_data_image.subwf.json]: This and the below var don't apply to this workflow.

File: daisy_integration_tests/create_random_data_image.subwf.json [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+          "Name": "inst-${NAME}",
+          "Disks": [{"Source": "disk-${NAME}"},{"Source": "disk-data-${NAME}-${ID}"}],
+          "MachineType": "n1-highcpu-4",
+          "Metadata": {
+            "block-project-ssh-keys": "true",
+	    "disk-size": "${size}",
+	    "disk-data-name": "disk-data-${NAME}-${ID}",
+	    "image-destination": "image-export-given-size-${size}gb"
+          },
+          "networkInterfaces": [
```
Comment (adjackura) [inline daisy_integration_tests/create_random_data_image.subwf.json]: You don't need the networkInterfaces setup

File: daisy_integration_tests/image_export_vmdk_given_size.subwf.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
+{
+  "Name": "image-export-vmdk-given-size-test",
+  "Vars": {
+    "size": {
+      "Required": true
+    },
+    "gcs_export": {
+      "Value": "${SCRATCHPATH}/created-by-export-test-given-size-${size}gb.vmdk"
```
Comment (adjackura) [inline daisy_integration_tests/image_export_vmdk_given_size.subwf.json]: You pass this var in in all the other workflows but you don't really need to, just have it in this workflow to reduce complexity, also just drop the size in the name, this path is fully scoped to this workflow by using SCRATCHPATH

File: daisy_integration_tests/image_export_vmdk_given_size.subwf.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    "verification_disk_name": "verify-export-${ID}"
+  },
+  "Steps": {
+    "export": {
+      "IncludeWorkflow": {
+        "Path": "../daisy_workflows/export/image_export_ext.wf.json",
+        "Vars": {
+          "destination": "${gcs_export}",
+          "export_instance_disk_image": "projects/compute-image-tools-test/global/images/family/debian-9-worker",
```
Comment (adjackura) [inline daisy_integration_tests/image_export_vmdk_given_size.subwf.json]: no need to set this var, just use the defaults

File: daisy_integration_tests/image_export_vmdk_given_size_verify.subwf.json [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    "create-tester": {
+      "CreateInstances": [
+        {
+          "disks": [{"Source": "disk-${NAME}-os"},{"Source": "${verification_disk_name}"}],
+          "name": "tester",
+          "metadata": {
+            "startup-script": "mkdir /mnt/sdb -p && sudo mount /dev/sdb /mnt/sdb && if ls /mnt/sdb/random.file 1> /dev/null 2>&1; then echo 'SUCCESS wVnWw3a41CVe3mBVvTMn'; else echo 'FAILURE wVnWw3a41CVe3mBVvTMn'; fi"
+          },
+          "Scopes": [
+            "https://www.googleapis.com/auth/devstorage.read_write"
```
Comment (adjackura) [inline daisy_integration_tests/image_export_vmdk_given_size_verify.subwf.json]: You shouldn't need read_write here

**Comment by dntczdx** [comment]
Removed all those redundant content as Andrew suggested.

**Comment by dntczdx** [comment]
Removed unused role and test pass :)

**Pull request merged**